File: src/Lista01/Questao01_02/dados/Veiculo.java
package Lista01.Questao01_02.dados;

import java.util.Objects;

public class Veiculo {
    private String modelo;
    private String marca;
    private String cor;
    private String placa;
    private Data devolucao;
    private int ano;

    public Veiculo() {
    }

    public Veiculo(String modelo, String marca, String cor, String placa, int ano, Data devolucao) {
        this.modelo = modelo;
        this.marca = marca;
        this.cor = cor;
        this.placa = placa;
        this.ano = ano;
        this.devolucao = devolucao;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getCor() {
        return cor;
    }

    public void setCor(String cor) {
        this.cor = cor;
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public int getAno() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public Data getDevolucao() {
        return devolucao;
    }

    public void setDevolucao(Data devolucao) {
        this.devolucao = devolucao;
    }

    public String toString() {
        return "Modelo: " + this.modelo + ", Marca: " + this.marca + ", Ano: " + this.ano + ", Cor: " + this.cor + ", Placa: " + this.placa + ", Data para devolução: " + this.devolucao.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Veiculo veiculo = (Veiculo) o;
        return ano == veiculo.ano && Objects.equals(modelo, veiculo.modelo) && Objects.equals(marca, veiculo.marca) && Objects.equals(cor, veiculo.cor) && Objects.equals(placa, veiculo.placa) && Objects.equals(devolucao, veiculo.devolucao);
    }

}

File: src/Lista01/Questao05/Veiculo.py
import string
from Data import Data

class Veiculo:

	def __init__(self, modelo, marca, cor, placa, ano, devolucao):
		self.__modelo = modelo
		self.__marca = marca
		self.__cor = cor
		self.__placa = placa
		self.__devolucao = devolucao
		self.__ano = ano
	
	def getModelo(self):
		return self.__modelo
	
	def setModelo(self, modelo: string):
		self.__modelo = modelo

	def getMarca(self):
		return self.__marca

	def setMarca(self, marca: string):
		self.__marca = marca

	def getCor(self):
		return self.__cor

	def setCor(self, cor: string):
		self.__cor = cor

	def getPlaca(self):
		return self.__placa

	def setPlaca(self, placa: string):
		self.__placa = placa

	def getAno(self):
		return self.__ano

	def setAno(self, ano: int):
		self.__ano = ano

	def getDevolucao(self):
		return self.__devolucao

	def setDevolucao(self, data: Data):
		self.__devolucao = data

	def __str__(self):
		return 'Modelo: {}, Marca: {}, Ano: {}, Cor: {}, Placa: {}, Data para Devolução: {}'.format(self.__modelo, self.__marca, self.__ano, self.__cor, self.__placa, self.__devolucao)